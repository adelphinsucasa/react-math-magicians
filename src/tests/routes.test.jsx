import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Root from '@/routes/root';
import '@testing-library/jest-dom/extend-expect';

describe('Test suite for the Routes components', () => {
  let topbar;
  beforeEach(() => {
    render(
      <BrowserRouter>
        <Root />
      </BrowserRouter>,
    );
    topbar = screen.getByText(/home/i);
  });

  it('Should test if routes (topbar) component is inside the document', () => {
    expect(topbar).toBeInTheDocument();
  });

  it('Should render a snapshot of the topbar component', () => {
    expect(topbar).toMatchSnapshot();
  });
});