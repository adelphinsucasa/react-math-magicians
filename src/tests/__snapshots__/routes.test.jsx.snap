// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test suite for the Routes components > Should render a snapshot of the topbar component 1`] = `
<a
  href="/home"
>
  Home
</a>
`;
